### ISSUE TITLE
feathers 4 with softDelete2 and feathers-mongoose fails with 404 not found

### ISSUE THREAD

**Issue by toovy** [feathersjs-ecosystem/feathers-hooks-common#532]
Hi, I'm currently upgrading to feathers 4.3.3. and my softDelete2 **before hook** in a **patch call** seems fail with a 404 not found.

Versions:

```
"@feathersjs/feathers": "4.3.3",
"feathers-hooks-common": "4.20.7",
"feathers-mongoose": "8.0.2",
```

In the before hook the constructed query contains `deletedAt: -1`. Since feathers 4.0 (see https://docs.feathersjs.com/guides/migrating.html#querying-by-id) the query is taken into consideration for patch calls, which might be the issue.

When looking at

https://github.com/feathers-plus/feathers-hooks-common/blob/master/lib/services/soft-delete-2.js#L174

one can see, that the query params are constructed with `deletedAt: -1` and in L179 this query is passed to the patch call. The **_patch method** of the **feathers-mongoose service** then first patches the record with `deletedAt: <timestamp>` and then tries to query the record with the ID and the query, that is, as one might recall, `deletedAt: -1`.

![Screenshot 2019-09-23 at 16 19 55](https://user-images.githubusercontent.com/350002/65436487-813ca380-de22-11e9-8132-5c09e6a3784f.png)

---

![Screenshot 2019-09-23 at 16 20 27](https://user-images.githubusercontent.com/350002/65436488-813ca380-de22-11e9-959f-1e85ee497606.png)

That means that first the record is set to `deletedAt: <timestamp>` and then it is queried with `deletedAt: -1`. So feathers-mongoose is not able to retrieve the record after patching it, because in my eyes the query is incorrect. But feathers-mongoose correctly returns a 404. I guess this 404 is not expected by softDelete2 and so the record is softdeleted, but the patch call to the service throws an error.

When removing the `[deletedAt]: -1` in L174 it works for me. But I don't know it that has any implications.

I'm not sure if this is the solution as this seems to be a general problem. Whenever you would use patch with a query, while one query parameter is modified with the patch, you would never find the resulting record again. So this might also be an issue with feathers-mongoose.

Any thoughts and suggestions?

**Comment by arfanliaqat** [comment]
Greetings, any update on this?

**Comment by MatthieuG9** [comment]
Hi ! I confirm the issue. Any solution ?

**Comment by MatthieuG9** [comment]
A small workaround, declare a function like this :
```ts
import { HookContext, Hook } from "@feathersjs/feathers";
import { checkContext } from "feathers-hooks-common";

export default function (_options: any = {}): Hook {
    return function FixSoftDelete404(ctx:HookContext<any>)
    {
        checkContext(ctx, 'error', 'remove');

        if(ctx.error && ctx.error.code == 404)
        {
            delete ctx.error;
            ctx.result = {message: 'ok'}
            ctx.statusCode = 200;
            return ctx;
        }
    }
};
```

And add it into hooks :
```ts
import FixSoftDelete404 from './fix-404';

export default {
  before: {
    all: [ authenticate('jwt'), softDelete2() ],
    find: [],
    get: [],
    create: [],
    update: [],
    patch: [],
    remove: []
  },

  after: {
    all: [ softDelete2() ],
    find: [],
    get: [],
    create: [],
    update: [],
    patch: [],
    remove: []
  },

  error: {
    all: [],
    find: [],
    get: [],
    create: [],
    update: [],
    patch: [],
    remove: [ FixSoftDelete404() ]
  }
};
```

**Comment by elaijuh** [comment]
I see master branch is updated, any plan for the new release with feathers v4 ?

**Comment by daffl** [comment]
This is now available as v5.0.x. New softDelete documentation at https://hooks-common.feathersjs.com/hooks.html#softdelete
